packages/nextalign/src/translate/translateGenes.cpp

```diff
@@ -115,7 +186,28 @@ PeptidesInternal translateGenes(                               //
     }
 
     auto& refGeneSeq = *extractRefGeneStatus.result;
+    const auto refGapCounts = countGaps(refGeneSeq);
+
     auto& queryGeneSeq = *extractQueryGeneStatus.result;
+    const auto queryGeneSize = safe_cast<int>(queryGeneSeq.size());
+
+    const auto queryGapCounts = countGaps(queryGeneSeq);
+    const bool queryIsAllGaps = queryGapCounts.total >= queryGeneSize;
+
+    // Handle the case when the query gene is completely missing
+    if (queryGeneSeq.empty() || queryIsAllGaps) {
+      const auto message = fmt::format(
+        "When processing gene \"{:s}\": The gene consists entirely from gaps. "
+        "Note that this gene will not be included in the results of the sequence.",
+        geneName);
+      warnings.inGenes.push_back(GeneWarning{.geneName = geneName, .message = message});
+      continue;
+    }
+
+    // NOTE: `+ 3` here is a magic number to give some additional space
+    const int bandWidth = safe_cast<int>(std::max(queryGapCounts.internal, refGapCounts.internal) / 3 + 3);
+    const int shift = queryGapCounts.leading + bandWidth / 2;
+    debug_trace("Deduced alignment params: bandWidth={:}, shift={:}\n", bandWidth, shift);
 
```

Comment:
This estimates bandwidth and shift here based on gap counts. We make sure that that empty peptides are discarded early, so that they don't cause problems.